modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -330,7 +330,12 @@ func (h *Handler) Cleanup() error {
 
 	// remove hosts from our config from the pool
 	for _, upstream := range h.Upstreams {
-		hosts.Delete(upstream.String())
+		// XXX: not sure if this will ever error - it's only possible if the
+		// value is a caddy.Destructor
+		_, err := hosts.Delete(upstream.String())
+		if err != nil {
+			return err
+		}
```

Comment:
(commented, but wanted to bring attention)

I'm not sure if this will ever actually error in this code-path - it's only possible if the value is a `caddy.Destructor`, and the only usage I found was in the root package. So, I could just do this instead:

```suggestion
		_, _ = hosts.Delete(upstream.String())
```